File: April-18/29-04-18/java/01-Little-jhool-and-psychic-powers.java

import java.io.*;


class TestClass {
    static boolean check(String str)
    {
        int n = str.length();
        int cnt1,cnt0;
        cnt1 = cnt0 = 0;
        
        for(int i = 0; i < n; i++)
        {
            if(str.charAt(i) == '0')
            {
                cnt0 += 1;
                cnt1 = 0;
            }
            else
            {
                cnt1 += 1;
                cnt0 = 0;
            }
            if(cnt1 == 6 || cnt0 == 6)
            {
                return true;
            }
        }
        return false;
    }
    
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        String str = input.readLine();
        if(!check(str))
            System.out.println("Good luck!");
        else
            System.out.println("Sorry, sorry!");
        
    }
}

File: April-18/29-04-18/python/01-Little-jhool-and-psychic-powers.py
def check(s):
    n = len(s)
    cnt0 = 0
    cnt1 = 0
    for i in range(0, n):
        if(s[i] == '0'):
            cnt0 += 1
            cnt1 = 0
        else:
            cnt0 = 0
            cnt1 += 1
        if(cnt0 == 6 or cnt1 == 6):
            return True
    return False
    
data = input()
if(check(data) == False):
    print("Good luck!")
else:
    print("Sorry, sorry!")
    
